### PR TITLE
Add last and init functions.  Add list of Coalton quirks to the intro doc.

### PR DESCRIPTION
**Pull request by HM0880**

### CODE REVIEW

File: docs/intro-to-coalton.md [hunk 1]
```diff
@@ -486,3 +486,9 @@ The following functions all take an optional package parameter
 * `print-value-db` - Print the type of every toplevel value
 * `print-class-db` - Print every class and their methods
 * `print-instance-db` - Print the instances of every class
+
+
+## Coalton quirks
```
Comment (stylewarning) [inline docs/intro-to-coalton.md:491]: "Quirks and Differences from Common Lisp"

File: docs/intro-to-coalton.md [hunk 2]
```diff
@@ -486,3 +486,9 @@ The following functions all take an optional package parameter
...
+
+- Coalton uses `True` (not `t`).
```
Comment (stylewarning) [inline docs/intro-to-coalton.md:493]: "As such, `t` can be used as an ordinary variable name."

File: docs/intro-to-coalton.md [hunk 3]
```diff
@@ -486,3 +486,9 @@ The following functions all take an optional package parameter
 * `print-value-db` - Print the type of every toplevel value
 * `print-class-db` - Print every class and their methods
 * `print-instance-db` - Print the instances of every class
+
+
+## Coalton quirks
+
```
Comment (stylewarning) [inline docs/intro-to-coalton.md:492]: "Lists in Coalton must be homogeneous."

"Coalton uses `fn`, *not* `lambda`, to denote anonymous functions."

"Numerical operators like `+` only take 2 arguments."

"Negation is done with `negate`. The form `(- x)` is a curried function equivalent to `(fn (z) (- x z))`."

File: src/library/list.lisp [hunk 4]
```diff
@@ -15,13 +15,26 @@
...
+    "Returns the last element of a list."
+    (match l
+      ((Cons x (Nil)) (Some x))
+      ((Cons _ xs) (last xs))
       ((Nil) None)))
 
+  (declare init ((List :a) -> (List :a)))
+  (define (init l)
+    "Returns every element except the last in a list."
+    (reverse (tail (reverse l))))
```
Comment (stylewarning) [inline src/library/list.lisp:36]: CL:BUTLAST could also be used

**Review by stylewarning — CHANGES_REQUESTED** [review]
Small requests and additions

File: src/library/list.lisp (hunk 4 above)
Comment (HM0880) [reply src/library/list.lisp:36]: Should I use CL:BUTLAST here?  I have no preference.

Comment (stylewarning) [reply src/library/list.lisp:36]: It will be about 2x as efficient, but it doesn't matter. (It'll just make one copy, not two.

**Review by stylewarning — APPROVED** [review]

**Pull request merged**